core_contracts/dex/dex.py

```diff
@@ -1481,7 +1495,7 @@ def loadBalancesAtSnapshot(self, _id: int, _snapshot_id: int, _limit: int, _offs
         if _offset < 0:
             revert(f"{TAG}: Offset must be equal to or greater than Zero.")
         rv = {}
-        for addr in self._active_addresses[_id].range(_offset, _offset + MAX_ITERATION_LOOP):
+        for addr in self._active_addresses[_id].range(_offset, _limit):
```

Comment:
Are you sure you didn't mean :

```for addr in self._active_addresses[_id].range(_offset, _offset + _limit):```

?

Reply:
good catch, pushed and rebased to address